globals/matchmaking_globals.go

```diff
@@ -99,3 +100,30 @@ func SearchGatheringWithMatchmakeSession(searchMatchmakeSession *match_making_ty
 	}
 	return returnSessionIndex
 }
+
+// SearchGatheringWithMatchmakeSession finds a gathering that matches with a MatchmakeSession
+func SearchGatheringWithSearchCriteria(lstSearchCriteria []*match_making_types.MatchmakeSessionSearchCriteria) uint32 {
+	var returnSessionIndex uint32 = 0
+
+	// This portion finds any sessions that match the search session. It does not care about anything beyond that, such as if the match is already full. This is handled below.
+	candidateSessionIndexes := make([]uint32, 0, len(Sessions))
+	for index, session := range Sessions {
+		if(len(lstSearchCriteria) == len(session.SearchCriteria)){
+			for criteriaIndex, criteria := range session.SearchCriteria {
+				if criteria.Equals(lstSearchCriteria[criteriaIndex]) {
+					candidateSessionIndexes = append(candidateSessionIndexes, index)
+				}
+			}
+		}
+	}
+	for _, sessionIndex := range candidateSessionIndexes {
+		sessionToCheck := Sessions[sessionIndex]
+		if len(sessionToCheck.ConnectionIDs) >= int(sessionToCheck.GameMatchmakeSession.MaximumParticipants) {
+			continue
+		} else {
+			returnSessionIndex = sessionIndex //found a match
+			break
+		}
+	}
```

Comment:
We also have to skip the sessions who have `OpenParticipation` disabled. This applies here and on `SearchGatheringWithMatchmakeSession`.

```suggestion
	for _, sessionIndex := range candidateSessionIndexes {
		sessionToCheck := Sessions[sessionIndex]
		if len(sessionToCheck.ConnectionIDs) >= int(sessionToCheck.GameMatchmakeSession.MaximumParticipants) {
			continue
		}
		
		if !sessionToCheck.GameMatchmakeExtension.OpenParticipation {
			continue
		}
		
		returnSessionIndex = sessionIndex // Found a match
		break
	}
```

This didn't affect on Mario Kart 7 before because the field was set to `true` by default.